src/user-lists/user-list-events-stats.service.ts

```diff
@@ -0,0 +1,324 @@
+import { DbUserListContributorStat } from "./entities/user-list-contributor-stats.entity";
+import {
+  UserListContributorStatsOrderEnum,
+  UserListContributorStatsTypeEnum,
+  UserListMostActiveContributorsDto,
+} from "./dtos/most-active-contributors.dto";
+import { ContributionPageMetaDto as ContributionsPageMetaDto } from "./dtos/contributions-pagemeta.dto";
+import { ContributionsPageDto } from "./dtos/contributions-page.dto";
+import { DbUserListContributor } from "./entities/user-list-contributor.entity";
+
+interface AllContributionsCount {
+  all_contributions: number;
+}
+
+@Injectable()
+export class UserListEventsStatsService {
+  constructor(
+    @InjectRepository(DbPullRequestGitHubEvents, "TimescaleConnection")
+    private pullRequestGithubEventsRepository: Repository<DbPullRequestGitHubEvents>,
+    @InjectRepository(DbUserListContributor, "ApiConnection")
+    private userListContributorRepository: Repository<DbUserListContributor>
+  ) {}
+
+  baseQueryBuilder(): SelectQueryBuilder<DbPullRequestGitHubEvents> {
+    const builder = this.pullRequestGithubEventsRepository.createQueryBuilder();
+
+    return builder;
+  }
+
+  private userListUsersQueryBuilder(): SelectQueryBuilder<DbUserListContributor> {
+    const builder = this.userListContributorRepository.createQueryBuilder("user_list_contributors");
+
+    return builder;
+  }
+
+  async findAllListContributorStats(
+    pageOptionsDto: UserListMostActiveContributorsDto,
+    listId: string
+  ): Promise<PageDto<DbUserListContributorStat>> {
+    const range = pageOptionsDto.range!;
+    const now = new Date().toISOString();
+
+    const userListUsersBuilder = this.userListUsersQueryBuilder();
+
+    userListUsersBuilder
+      .leftJoin("users", "users", "user_list_contributors.user_id=users.id")
+      .where("user_list_contributors.list_id = :listId", { listId });
+
+    const allUsers = await userListUsersBuilder.getMany();
+
+    if (allUsers.length === 0) {
+      return new ContributionsPageDto(
+        new Array<DbUserListContributorStat>(),
+        new ContributionsPageMetaDto({ itemCount: 0, pageOptionsDto }, 0)
+      );
+    }
+
+    const users = allUsers.map((user) => (user.login ? user.login.toLowerCase() : user.username));
+
+    const userListQueryBuilder =
+      this.pullRequestGithubEventsRepository.manager.createQueryBuilder() as SelectQueryBuilder<DbPullRequestGitHubEvents>;
+
+    userListQueryBuilder.select("DISTINCT users.login", "login");
+
+    userListQueryBuilder
+      .addSelect(
+        `(SELECT COALESCE(sum(push_num_commits), 0)
+          FROM push_github_events 
+          WHERE LOWER(actor_login)=users.login
+          AND push_ref IN ('refs/heads/main', 'refs/heads/master')
+          AND now() - INTERVAL '${range} days' <= event_time)::INTEGER`,
+        "commits"
+      )
+      .addSelect(
+        `(SELECT COALESCE(count(event_id), 0)
+        FROM pull_request_github_events 
+        WHERE LOWER(actor_login)=users.login
```

Comment:
is `users.login` normalized to be lower? If not we should add:

```suggestion
        WHERE LOWER(actor_login)=LOWER(users.login)
```

Reply:
Yep, `users.login` is normalized https://github.com/open-sauced/api/pull/476/files#diff-6100f173368be97e4741b42b59e6e47d26e93fcc9d3278258363204d42ac0685R129